### ISSUE TITLE
[Accessibility Bug] - Responsive Vertical Menu - Interactive elements outside of the open menu should not be usable at desktop view

### ISSUE THREAD

**Issue by amieedwards-sage** [Sage/carbon#7579]
### Description

When the Responsive Vertical Menu is open, the background is still enabled and buttons/links/hover states can be activated. This happens for the 700px breakpoint and above.

This was previously raised as an issue and closed: https://github.com/Sage/carbon/issues/7483
However feedback from Accessibility have confirmed that this is still an issue and have provided suggestions below.

### Reproduction

https://stackblitz.com/edit/parsium-carbon-starter-jchqovxf?file=src%2FApp.tsx

### Steps to reproduce

Open Responsive Vertical Menu on a screen over 700px and click the button.

### JIRA ticket numbers (Sage only)

SBS-142269

### Suggested solution

Advice from Accessibility is that at full desktop mode, the Responsive Vertical Menu should act like a non lightbox modal (aka, a modal that doesn't have a dimmed background). This means interactive elements shouldn't be able to be interacted with at this point.

Recommendation: treat as modal, clicking outside of the Responsive Vertical Menu should close the popover, then items on page can be interacted with again.

### Carbon version

157.1.1

### React version

v17

### Design tokens version

4.34.0

### Relevant browsers

Chrome

### Relevant OSs

MacOS

### Confidentiality

- [x] I confirm there is no confidential or commercially sensitive information included.
